warehouse/query-core/src/main/java/datawave/query/jexl/functions/EventFieldAggregator.java

```diff
@@ -122,16 +112,48 @@ private Set<String> getNormalizedValues(String dataType, String fieldName, Strin
 
         // transform the key for each type and add it to the normalized set
         for (String typeClass : typeClasses) {
-            try {
-                Type<?> type = typeCache.get(typeClass);
-                String normalizedValue = type.normalize(fieldValue);
+            Type<?> type = getTypeFactory().createType(typeClass);
+            String normalizedValue = type.normalize(fieldValue);
 
-                normalizedValues.add(normalizedValue);
-            } catch (ExecutionException e) {
-                throw new RuntimeException("cannot instantiate class '" + typeClass + "'", e);
-            }
+            normalizedValues.add(normalizedValue);
         }
 
         return normalizedValues;
     }
+
+    /**
+     * Get the TypeFactory. If no TypeFactory exists one will be created. Configs for cache size and timeout may be configured.
+     *
+     * @return the TypeFactory
+     */
+    private TypeFactory getTypeFactory() {
+        if (typeFactory == null) {
+            if (typeCacheSize != -1 && typeCacheTimeoutMinutes != -1) {
```

Comment:
What if one of the 2 is set?

Reply:
Then a type factory with default expiration values will be created.